### DISCUSSION TITLE
使用Mishards在k8s上用helm部署milvus，出现pod状态是CrashLoopBackOff

### DISCUSSION THREAD

**Discussion by tom-lch** [milvus-io/milvus#4599] [Q&A and General discussion]
![image](https://user-images.githubusercontent.com/49295186/104261690-10ab1180-54c1-11eb-89cc-1cd2111006b5.png)

![image](https://user-images.githubusercontent.com/49295186/104261621-ea857180-54c0-11eb-859b-121421b5744c.png)
![image](https://user-images.githubusercontent.com/49295186/104261663-fec96e80-54c0-11eb-8248-54528e3896b6.png)

**Comment by jeffoverflow** [comment]
Can you execute the follow commands to observe the results？

`kubectl logs milvus-release-readonly-6d54fbfc8d-thxq7 -n milvus`

**Reply by jeffoverflow** [reply]
Currently, I don't see the status of MySQL server.

**Reply by tom-lch** [reply]
I use TIDB deployed in k8s as externalMysql, so didn't show mysql in this photo.
but now I use mysql also show this question.
![image](https://user-images.githubusercontent.com/49295186/104289443-1bcb6500-54f4-11eb-92e6-2b0fb910fb72.png)
`
Name:         milvus-release-readonly-5b9ffc4f8b-46kw9
Namespace:    milvus
Priority:     0
Node:         k8s-node3/192.168.1.103
Start Time:   Tue, 12 Jan 2021 16:24:08 +0800
Labels:       app.kubernetes.io/instance=milvus-release
              app.kubernetes.io/name=milvus
              component=readonly
              pod-template-hash=5b9ffc4f8b
Annotations:  checksum/config: 3cfc30059c378f98e60ad3d398f7a3bea442418bc1f249014938716a6045841f
Status:       Running
IP:           10.244.3.37
IPs:
  IP:           10.244.3.37
Controlled By:  ReplicaSet/milvus-release-readonly-5b9ffc4f8b
Init Containers:
  wait-for-mysql:
    Container ID:  docker://c5cce7e55d2d50fa9b946cdf3d83a71b229b20bb993deb706bee35253a851d2c
    Image:         alpine:3.8
    Image ID:      docker-pullable://alpine@sha256:2bb501e6173d9d006e56de5bce2720eb06396803300fe1687b58a7ff32bf4c14
    Port:          <none>
    Host Port:     <none>
    Command:
      sh
      -c
      until nc -z -w 2 milvus-release-mysql 3306 && echo mysql database ok; do
        sleep 2;
      done;

    State:          Terminated
      Reason:       Completed
      Exit Code:    0
      Started:      Tue, 12 Jan 2021 16:24:10 +0800
      Finished:     Tue, 12 Jan 2021 16:25:30 +0800
    Ready:          True
    Restart Count:  0
    Environment:    <none>
    Mounts:
      /var/run/secrets/kubernetes.io/serviceaccount from default-token-qdgws (ro)
  check-for-share-storage:
    Container ID:  docker://7654d5e337a2bdcfa703b0c93a7b7b90a28405d8a537c9dc40b694a15cc91bef
    Image:         alpine:3.8
    Image ID:      docker-pullable://alpine@sha256:2bb501e6173d9d006e56de5bce2720eb06396803300fe1687b58a7ff32bf4c14
    Port:          <none>
    Host Port:     <none>
    Command:
      sh
      -c
      until test -f /var/lib/milvus/db/test/milvus-release.txt && echo read from share storage; do
        sleep 2;
      done;

    State:          Terminated
      Reason:       Completed
      Exit Code:    0
      Started:      Tue, 12 Jan 2021 16:25:32 +0800
      Finished:     Tue, 12 Jan 2021 16:25:32 +0800
    Ready:          True
    Restart Count:  0
    Environment:    <none>
    Mounts:
      /var/lib/milvus/db from milvus-data-disk (ro)
      /var/run/secrets/kubernetes.io/serviceaccount from default-token-qdgws (ro)
Containers:
  milvus:
    Container ID:   docker://e47a2d98ee0de68ec344b53467945c24887282f87108d94a49cbe4467392d44c
    Image:          milvusdb/milvus:0.10.5-cpu-d010621-4eda95
    Image ID:       docker-pullable://milvusdb/milvus@sha256:b87bf94109e85193be5a1b4f7b1d0b80a169a0cdcae348d49daf455da1ae08df
    Ports:          19530/TCP, 19121/TCP
    Host Ports:     0/TCP, 0/TCP
    State:          Waiting
      Reason:       CrashLoopBackOff
    Last State:     Terminated
      Reason:       Error
      Exit Code:    132
      Started:      Tue, 12 Jan 2021 16:31:16 +0800
      Finished:     Tue, 12 Jan 2021 16:31:16 +0800
    Ready:          False
    Restart Count:  6
    Liveness:       http-get http://:19121/system/status delay=30s timeout=10s period=15s #success=1 #failure=5
    Readiness:      http-get http://:19121/system/status delay=15s timeout=10s period=15s #success=1 #failure=3
    Environment:    <none>
    Mounts:
      /var/lib/milvus/conf/server_config.yaml from milvus-configmap (rw,path="readonly_server_config.yaml")
      /var/lib/milvus/db from milvus-data-disk (rw)
      /var/lib/milvus/logs from milvus-logs-disk (rw)
      /var/run/secrets/kubernetes.io/serviceaccount from default-token-qdgws (ro)
Conditions:
  Type              Status
  Initialized       True
  Ready             False
  ContainersReady   False
  PodScheduled      True
Volumes:
  milvus-configmap:
    Type:      ConfigMap (a volume populated by a ConfigMap)
    Name:      milvus-release
    Optional:  false
  milvus-data-disk:
    Type:       PersistentVolumeClaim (a reference to a PersistentVolumeClaim in the same namespace)
    ClaimName:  milvus-release
    ReadOnly:   false
  milvus-logs-disk:
    Type:       PersistentVolumeClaim (a reference to a PersistentVolumeClaim in the same namespace)
    ClaimName:  milvus-release-logs
    ReadOnly:   false
  default-token-qdgws:
    Type:        Secret (a volume populated by a Secret)
    SecretName:  default-token-qdgws
    Optional:    false
QoS Class:       BestEffort
Node-Selectors:  <none>
Tolerations:     node.kubernetes.io/not-ready:NoExecute for 300s
                 node.kubernetes.io/unreachable:NoExecute for 300s
Events:
  Type     Reason     Age                   From                Message
  ----     ------     ----                  ----                -------
  Normal   Scheduled  10m                   default-scheduler   Successfully assigned milvus/milvus-release-readonly-5b9ffc4f8b-46kw9 to k8s-node3
  Normal   Pulled     10m                   kubelet, k8s-node3  Container image "alpine:3.8" already present on machine
  Normal   Created    10m                   kubelet, k8s-node3  Created container wait-for-mysql
  Normal   Started    10m                   kubelet, k8s-node3  Started container wait-for-mysql
  Normal   Pulled     9m2s                  kubelet, k8s-node3  Container image "alpine:3.8" already present on machine
  Normal   Created    9m2s                  kubelet, k8s-node3  Created container check-for-share-storage
  Normal   Started    9m2s                  kubelet, k8s-node3  Started container check-for-share-storage
  Normal   Pulled     8m20s (x4 over 9m1s)  kubelet, k8s-node3  Container image "milvusdb/milvus:0.10.5-cpu-d010621-4eda95" already present on machine
  Normal   Created    8m20s (x4 over 9m1s)  kubelet, k8s-node3  Created container milvus
  Normal   Started    8m20s (x4 over 9m)    kubelet, k8s-node3  Started container milvus
  Warning  BackOff    12s (x46 over 8m59s)  kubelet, k8s-node3  Back-off restarting failed container

`
